redhat-cop/tower_utilities
# add aap-setup-download role to download automatically the AAP setup file

### What does this PR do?

Adds a role to automatically download the installer image from RHN

### How should this be tested?

My playbook looks so far like this:

```yaml
- name: download and install AAP from the bastion
  hosts: bastion.*
  gather_facts: false
  become: true
  tags: aap_installation
  roles:
    - redhat_cop.tower_utilities.aap_setup_download
  post_tasks:
    - debug:
        var: aap_setup_down_installer_file
```
And the only variable I really need to define in my vault is aap_setup_down_offline_token (see `defaults/main.yml` for the details and other variables).

### Is there a relevant Issue open for this?

n/a

### Other Relevant info, PRs, etc.

Adapted from https://www.redhat.com/en/blog/automating-installation-ansible-automation-platform-ansible-and-satellite

I still need to work on the "soft" aspects of the role, means documentation :-) hence still WiP, else it works good enough for me.
